plugins/http_plugin/include/eosio/http_plugin/beast_http_session.hpp

```diff
@@ -251,13 +252,13 @@ class beast_http_session : public detail::abstract_conn,
    // shared_from_this() requires default constructor
    beast_http_session() = default;
 
-   beast_http_session(Socket&& socket, http_plugin_state& plugin_state, std::string remote_endpoint,
+   beast_http_session(Socket&& socket, std::shared_ptr<http_plugin_state> plugin_state, std::string remote_endpoint,
                       api_category_set categories, const std::string& local_address)
-       : socket_(std::move(socket)), categories_(categories), plugin_state_(plugin_state),
+       : plugin_state_(plugin_state), socket_(std::move(socket)), categories_(categories),
```

Comment:
This can be `std::move(plugin_state)`

Reply:
I usually always copy `shared_ptr`s, and it feels weird to move something in the initializer list that isn't (and can't be) moved in the ctor call,
https://github.com/AntelopeIO/leap/blob/5fcf5b49548e58e8bb01b50b6d7f2a9458602b0f/plugins/http_plugin/http_plugin.cpp#L230-L232

But the code generation is surprisingly better when adding the `move` here, so maybe it's worth doing

Reply:
Move of `shared_ptr` is nice as it avoid the `std::atomic` operation I believe.